### DISCUSSION TITLE
terminalGPT

### DISCUSSION THREAD

**Discussion by SuperElectron** [ohmyzsh/ohmyzsh#11434] [Help]
Hello friends!

So I love oh-my-zsh and have been using it for a while.  I recently came across this: https://www.producthunt.com/products/terminalgpt

I am not very familiar with how to write a custom plugin for oh-my-zsh, I tried a few things, but had no success. Would anyone be intested with helping get terminalGPT integrated with oh-my-zsh as a custom plugin?  I believe it would be MEGA powerful to add to oh-my-zsh if its possible.  I'd love to hear feedback and get an understanding of this possibility.

resources: 

1. https://github.com/jucasoliveira/terminalGPT?ref=producthunt
2. https://github.com/darrenbutcher/plugin
3. https://github.com/ohmyzsh/ohmyzsh/wiki/Customization#adding-a-new-plugin

**Comment by robbyrussell** [comment]
I sent out a [call on Twitter](https://twitter.com/ohmyzsh/status/1612940788635422722) to see if anyone is interested in helping you out!

-Robby

**Reply by jsamuel1** [reply]
Something like https://github.com/shorwood/gptsh would he interesting to consider here too - creates and executes shell commands from natural language.

**Comment by ferblape** [comment]
I'd like to suggest this resource for some inspiration https://github.com/Aesthetikx/openai_pipe

**Comment by ebesson** [comment]
Something like would he interesting to consider https://github.com/jseguillon/acli : Assisted command line interface powered by OpenAI's API. This cli use spf13/cobra, this framework is able to generate autocomplete for zsh : https://github.com/spf13/cobra/blob/main/shell_completions.md .

**Comment by jseguillon** [comment]
Here to help if you adopt acli 🙂
